data-reconciliation-app/src/auth/Jwt-validation.ts

```diff
@@ -0,0 +1,42 @@
+import * as ccfapp from "@microsoft/ccf-app";
+import { ServiceResult } from "../utils/service-result";
+import msIdProvider from "./jwtIssuer/MS-AAD";
+import testIdProvider from "./jwtIssuer/TestIdP";
+
+/**
+ * JWT Identity Providers
+*/
+export enum JwtIdentityProviderEnum {
+    MS_AAD = "https://login.microsoftonline.com/common/v2.0",
+    Test = "https://demo"
+}
+type identityId = string;
+
+export interface IJwtIdentityProvider {
+    isValidJwtToken(identity: ccfapp.JwtAuthnIdentity): ServiceResult<identityId>
+}
+
+export interface IValidatorService {
+    validate(request: ccfapp.Request<any>): ServiceResult<identityId>
+}
+
+export class JwtValidator implements IValidatorService  {
+    private readonly identityProviders = new Map<JwtIdentityProviderEnum, IJwtIdentityProvider>();
+
+  constructor() {
+    this.identityProviders.set(JwtIdentityProviderEnum.MS_AAD, msIdProvider);
```

Comment:
As this validator is implemented by both `MS_AAD` and `Test`, maybe we can put them under `/auth/validator/jwt/`?

I imagine it could be:

```
- auth
   - validator
       - validaton-service.ts
       - jwt
           - jwt-validator.ts
           - ms-aad-jwt-validator.ts
           - test-jwt-validator.ts
```